incubator/hnc/internal/kubectl/describe.go

```diff
@@ -67,34 +74,72 @@ var describeCmd = &cobra.Command{
 			fmt.Printf("  No children\n")
 		}
 
-		// Early exit if no conditions
+		// Conditions
 		if len(hier.Status.Conditions) == 0 {
 			fmt.Printf("  No conditions\n")
-			return
-		}
-
-		// Conditions
-		fmt.Printf("  Conditions:\n")
-		for _, c := range hier.Status.Conditions {
-			fmt.Printf("  - %s: %s\n", c.Code, c.Msg)
-			if len(c.Affects) == 0 {
-				continue
-			}
-			fmt.Printf("    Affected by this condition:\n")
-			for _, a := range c.Affects {
-				if a.Name != "" {
-					if a.Group == "" {
-						a.Group = "core"
+		} else {
+			fmt.Printf("  Conditions:\n")
+			for _, c := range hier.Status.Conditions {
+				fmt.Printf("  - %s: %s\n", c.Code, c.Msg)
+				if len(c.Affects) == 0 {
+					continue
+				}
+				fmt.Printf("    Affected by this condition:\n")
+				for _, a := range c.Affects {
+					if a.Name != "" {
+						if a.Group == "" {
+							a.Group = "core"
+						}
+						fmt.Printf("      - %s/%s (%s/%s/%s)\n", a.Namespace, a.Name, a.Group, a.Version, a.Kind)
+					} else {
+						fmt.Printf("      - %s\n", a.Namespace)
 					}
-					fmt.Printf("      - %s/%s (%s/%s/%s)\n", a.Namespace, a.Name, a.Group, a.Version, a.Kind)
-				} else {
-					fmt.Printf("      - %s\n", a.Namespace)
 				}
 			}
 		}
```

Comment:
This is a good time to move this to its own function too rather than have a very deeply nested code.

Reply:
Done